pyquil/api/qpu.py

```diff
@@ -13,27 +13,99 @@
 #    See the License for the specific language governing permissions and
 #    limitations under the License.
 ##############################################################################
+import warnings
 
+import requests
 from six import integer_types
 
 from pyquil.quil import Program
 from ._base_connection import validate_run_items, TYPE_MULTISHOT, TYPE_MULTISHOT_MEASURE, get_job_id, BaseConnection
+from ._config import PyquilConfig
+
+
+def get_devices(async_endpoint='https://job.rigetti.com/beta', api_key=None, user_id=None):
+    """
+    Get a list of currently available devices. The arguments for this method are the same as those for QPUConnection.
+    Note that this method will only work for accounts that have QPU access.
+
+    :return: list of online and offline devices
+    :rtype: list
+    """
+    config = PyquilConfig()
+    api_key = api_key if api_key else config.api_key
+    user_id = user_id if user_id else config.user_id
+
+    headers = {
+        'X-Api-Key': api_key,
+        'X-User-Id': user_id,
+        'Content-Type': 'application/json; charset=utf-8'
+    }
+
+    response = requests.get(async_endpoint + '/devices', headers=headers)
+    return {Device(name, device) for (name, device) in response.json()['devices'].items()}
+
+
+class Device(object):
+    """
+    A device (quantum chip) that can accept programs. Only devices that are online will actively be accepting new
+    programs.
+    """
+    def __init__(self, name, raw):
+        """
+        :param name: name of the device
+        :param raw: raw JSON response from the server with additional information about this device
+        """
+        self.name = name
+        self.raw = raw
+
+    def is_online(self):
+        """
+        Whether or not the device is online and accepting new programs.
+        :rtype: bool
```

Comment:
need a newline between description and parameter/return docstring lines for sphinx rendering